File: src/main/java/org/customer/constants/AppConstants.java
package org.customer.constants;

public final class AppConstants {

    private AppConstants() {}


  public static final String GROUP_ID = "group_id";

  public static final double THRESHOLD_D = 0.0;

  public static final int THRESHOLD = 0;

  public static final String  ACCOUNT_TOPIC_NAME = "topic-account";

  public static final String  TRANSACTION_TOPIC_NAME = "topic-transaction";

  public static final String DATE_FORMAT = "yyyyMMddHHmmss";

  public static  final Integer FACTORY_CONCURRENCY_COUNT = 3;

  public static  final int MAX_RETRY_ATTEMPT = 3;

  public static  final long IDLE_BETWEEN_POLLS = 200000;

  public static  final int MAX_POLL_INTERVAL_MS_CONFIG = 600000;

  public static  final int SESSION_TIMEOUT_MS_CONFIG = 50000;

  public static  final int DEFAULT_API_TIMEOUT_MS_CONFIG = 7000000;


  public static  final int MAX_POLL_RECORDS_CONFIG = 100;

  public static  final String GROUP_INSTANCE_ID_CONFIG = "kafka-streams-instance-1";

  public static  final int MAX_PARTITION_FETCH_BYTES_CONFIG = 10000;

  public static  final int FETCH_MAX_BYTES_CONFIG = 30000;

  public static  final long CACHE_TTL = 60000;

  public static final String BOOTSTRAP_ADDRESS = "localhost:9092";

  public static final String CONSUMER_GROUP_ID = "my-group" ;



}

File: src/main/java.kotlin/org/customer/constants/AppConstants.kt
package org.customer.constants

object AppConstants {
    const val GROUP_ID: String = "group_id"

    const val THRESHOLD_D: Double = 0.0

    const val THRESHOLD: Int = 0

    const val ACCOUNT_TOPIC_NAME: String = "topic-account"

    const val TRANSACTION_TOPIC_NAME: String = "topic-transaction"

    const val DATE_FORMAT: String = "yyyyMMddHHmmss"

    const val FACTORY_CONCURRENCY_COUNT: Int = 3

    const val MAX_RETRY_ATTEMPT: Int = 3

    const val IDLE_BETWEEN_POLLS: Long = 200000

    const val MAX_POLL_INTERVAL_MS_CONFIG: Int = 600000

    const val SESSION_TIMEOUT_MS_CONFIG: Int = 50000

    const val DEFAULT_API_TIMEOUT_MS_CONFIG: Int = 7000000


    const val MAX_POLL_RECORDS_CONFIG: Int = 100

    const val GROUP_INSTANCE_ID_CONFIG: String = "kafka-streams-instance-1"

    const val MAX_PARTITION_FETCH_BYTES_CONFIG: Int = 10000

    const val FETCH_MAX_BYTES_CONFIG: Int = 30000

    const val CACHE_TTL: Long = 60000

    const val BOOTSTRAP_ADDRESS: String = "localhost:9092"

    const val CONSUMER_GROUP_ID: String = "my-group"
}